data_structures/binary_tree/binary_search_tree.py

```diff
@@ -69,12 +79,12 @@ def __insert(self, value):
                         parent_node = parent_node.right
             new_node.parent = parent_node
 
-    def insert(self, *values):
+    def insert(self, *values) -> BinarySearchTree:
```

Comment:
Postponed evaluation of annotations has been put on hold indefinitely, so we should not be incorporating them into the codebase until the PEP is finalized: https://docs.python.org/3.11/whatsnew/3.11.html#pep-563-may-not-be-the-future. Please update this and all the similar changes.